Solutions/CohesitySecurity/Tools/configure_permissions.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/zsh
+
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+#
+# Description: This script assigns Microsoft Sentinel Automation Contributor role
+# permissions to the specified app display name and role name.
+# It first retrieves the principal ID and role definition ID, then
+# grants the necessary permissions for the playbook.
+
+app_display_name="Azure Security Insights"
+role_name="Microsoft Sentinel Automation Contributor"
+
+# Call the Python script to get the principal ID
+principal_id=$(python3 ./get_principal_id.py "$app_display_name")
+error_handler "Failed to get the principal ID."
+
+# Call the script to get the role definition ID
+role_definition_id=$(./get_role_definition_id.sh "$role_name")
+error_handler "Failed to get the role definition ID."
+
+# Print resource_group and workspace_name for later debug
+echo "During configure permissions, Resource Group: $resource_group"
+echo "During configure permissions, Workspace Name: $workspace_name"
+
+# If the principal ID and role definition ID were found, proceed with the rest of the script
+if [[ ! "$principal_id" =~ "No service principal found" ]] && [[ ! "$role_definition_id" =~ "No role definition found" ]]; then
+    echo "Principal ID: $principal_id"
+    echo "Role Definition ID: $role_definition_id"
+    role_definition_id="/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleDefinitions/$role_definition_id"
+
+    # Set the subscription
+    az account set --subscription "$subscription_id"
+    error_handler "Failed to set the subscription."
+
+    # Grant the necessary permissions for the playbook
+    sentinel_resource_id=$(az monitor log-analytics workspace show --resource-group "$resource_group" --workspace-name "$workspace_name" --query 'id' -o tsv)
+    error_handler "Failed to get the Sentinel resource ID."
+
+    # Set role assignment ID
+    role_assignment_id=$(uuidgen)
+    api_version="2015-07-01"
+
+    # Create the request body
+    request_body=$(jq -n \
+        --arg roleDefinitionId "$role_definition_id" \
+        --arg principalId "$principal_id" \
+        '{"properties": {"roleDefinitionId": $roleDefinitionId, "principalId": $principalId}}'
+    )
+
+    # Send a batch request
+    az rest --method POST \
```

Comment:
I'm trying to find the difference between this file and the grant_role.sh. My understanding is that both of these scripts are doing the same thing. Could you please help me explain the difference?


Btw, we should directly post the request to https://management.azure.com/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleAssignments/$role_assignment_id?api-version=$api_version

I don't see a reason for using 'https://management.azure.com/batch' here.

Reply:
they are different.
grant_role.sh is assign role to user. And it is a done through az cli. Which means, it should be supported by MS for a long time. 
configure_permissions.sh is to add playbook permission to workspace. This way is like hack.
I got https://management.azure.com/batch through chrome console, and mimic the similar request, that's why I said it is a hack.

what's your reason for using https://management.azure.com/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleAssignments/$role_assignment_id?api-version=$api_version ?

Reply:
It is no longer a hack if we remove the 'batch' REST API and directly use the role_assignment REST API. The reason for using the API I mentioned directly is that 'batch' is not a 'documented' REST API call, but the roleAssignments REST API is a documented and support REST API call.

I'm not sure if you have got a chance to explore, but there may just be an az cli command to perform this just as you have done in grant_role.sh.

Reply:
Please note that this process does not involve role assignment. Instead, it follows the manual configuration workflow found in the Azure Portal under Microsoft Sentinel Workspace > Settings > Playbook permissions.

Despite extensive research, I have been unable to find a corresponding Azure CLI command for this specific task. It is possible that a relevant Azure CLI command was available in the past but has since been removed. As a result, some older posts may reference this functionality, but none of the suggested solutions are currently functional.

Reply:
"Please note that this process does not involve role assignment"
> Why do you say that? If you look at the REST API, it doing exactly that. Look for the string 'roleAssignments'


Also, you didn't answer regarding the use of the official support REST API. I provided the URL.